# Задача

Приток новых игроков делится на 2 основных источника - рекламный и органический. Отдел маркетинг выдвигает гипотезу о связи рекламного и органического траффика: когда маркетологи привлекают больше рекламного траффика, растет и органический.

Для выполнения задачи необходимо использовать тестовый датасет (таблица source_comparison).

- Подтвердите или опровергните гипотезу маркетологов. Если вы ее опровергаете, то сформулируйте гипотезу которая кажется вам наиболее вероятной и проверьте ее
- Количественно рассчитайте вероятность что гипотеза маркетологов (или ваша) верна

# Решение

In [15]:
import sqlite3
import pandas as pd
import sqlalchemy

In [3]:
# подключимся к тестовой бд
connect = sqlite3.connect("testcase.db")

In [16]:
# переведем в датафрейм
df = pd.read_sql_query("SELECT * FROM source_comparison", connect)

In [18]:
def get_dataset_info(df: pd.DataFrame) -> None:
    """
    Функция выводит общую инфу
    о датасете одним полем
    
    :param pd.DataFrame df: pandas-датафрейм, который нужно проанализировать
    """
    display(df.head())
    print('\n')
    print(df.info())
    print('\n')
    display(df.describe())
    print('\n')
    print(df.isna().sum())
    print('\n')
    print(f'Дубликаты: {df.duplicated().sum()}')
    print('it`s over')

In [19]:
# получим общую инфу о представленных данных
get_dataset_info(df)

Install_Dates source_type Country  installs
0    2020-05-18        Paid      PY       4.0
1    2020-04-14        Paid      FR      35.0
2    2020-04-30        Paid      JP      25.0
3    2020-03-20        Paid      DE      11.0
4    2020-04-30        Paid      IT       8.0



<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18606 entries, 0 to 18605
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Install_Dates  18606 non-null  object 
 1   source_type    18606 non-null  object 
 2   Country        18606 non-null  object 
 3   installs       18606 non-null  float64
dtypes: float64(1), object(3)
memory usage: 581.6+ KB
None




installs
count  18606.000000
mean      51.968451
std      227.810616
min        1.000000
25%        2.000000
50%        6.000000
75%       22.000000
max     5633.000000



Install_Dates    0
source_type      0
Country          0
installs         0
dtype: int64


Дубликаты: 679
it`s over


Вроде проблем никаких нет, за исключением 679 полных дубликата. Дубликаты в целом не очень хорошо, поэтому от них избавимся, чтобы не смазали картину.

In [21]:
df = df.drop_duplicates()
df.duplicated().sum()

0

Нас в первую очередь интересуют столбцы с видом трафика и количеством установок.